### ISSUE TITLE
support opening text file for precommand

### ISSUE THREAD

**Issue by heartacker** [microsoft/terminal#16495]
# Description of the new feature/enhancement

For many developers, especially embedded developers, they would pre-define many text commands and save them in a text file. If WT can split views and open and display text in one view, we can quickly send the text commands to the terminal. I can send lines, send selected blocks, or send the entire content of the file, and preset some shortcuts.

# Proposed technical implementation details (optional)

![image](https://github.com/microsoft/terminal/assets/1876302/2be7c2e4-7323-429c-b31b-58942120dec6)

**Comment by zadjii** [comment]
Guys this is **totally** the notebooks thing I was talking about 😉 


But also, there's probably some other useful things we're working on here for "Tasks" - basically, having a big long list of `sendInput` actions that you can quickly activate - either through the suggestions UI, or maybe even adding a "Tasks" pane. 

(we'll probably have more to say after the holidays)

**Comment by zadjii-msft** [comment]
Okay so, you may be interested in the Suggestions UI and Tasks, which are new in 1.19: [Using SendInput Actions in the Suggestions UI](https://devblogs.microsoft.com/commandline/windows-terminal-preview-1-19-release/#using-sendinput-actions-in-the-suggestions-ui)

![](https://devblogs.microsoft.com/commandline/wp-content/uploads/sites/33/2023/09/SuggestionsUI.gif)

I have Big Plans for these in upcoming releases - some of which is tracked in #13445 (though I think a better thread is floating around here somewhere)

see also:
* #1595
* #15664

**Comment by heartacker** [comment]
no, I know this, and we DO NOT want to config the setting.json, we just want to code in some text file for TeamWorks

so, we want to open the text and the send those command easily, through terminal

**Comment by zadjii-msft** [comment]
No worries! I've got a lot of related ideas in this space, and untangling them all is sometimes a bit tricky.

You're looking for something more like this?
![](https://raw.githubusercontent.com/microsoft/terminal/dev/migrie/s/north-star/doc/specs/NorthStar/img/mockup-000.png)

Yea? A file, with blocks of code that lets you run individual commands, or the whole block?

**Comment by heartacker** [comment]
This is so innovative.
and  a little simple txt file for me  
We can run the line to terminal where curser focused (with  shortcut )
or  select  some block  to  run

**Comment by zadjii-msft** [comment]
Congrats, this gets to be the thread then ☺️ 

I've been ideating in this space for a long time, just never actually filed a thread for it. I'm gonna co-opt the OP here to add some other notes. Thanks!

**Comment by heartacker** [comment]
@zadjii-msft 
here is some demo using vscode.(yes, we can do this with vscode)
1. text base panel is need indeed (good for teamwork and easy to use)
2. notebook panel is welcome also (good for guiding)

![txt notebook_panel](https://github.com/microsoft/terminal/assets/1876302/ca2797de-5a12-4c11-8fef-bd396a46d332)

**Comment by zadjii-msft** [comment]
_reserving for notes for hackathon_

**Comment by zadjii-msft** [comment]
I'm gonna leave the above comment for the "inline notebooks" investigation that eventually petered out, and use this one for side-by-side markdown in the Terminal

Branch map:

* `dev/migrie/f/sui-panes`
  * + til events ==> `dev/migrie/fhl/2024-spring-merge-base`
    * **Inline notebooks**: https://github.com/microsoft/terminal/compare/dev/migrie/fhl/2024-spring-merge-base...dev/migrie/fhl/2024-inline-notebook
  * Tasks pane: https://github.com/microsoft/terminal/compare/dev/migrie/f/sui-panes...dev/migrie/fhl/tasks-pane
    * **Markdown pane**: https://github.com/microsoft/terminal/compare/dev/migrie/f/tasks-pane...dev/migrie/fhl/md-pane



_2024-march-14 EoD screenshot_: 
![image](https://github.com/microsoft/terminal/assets/18356694/204f83af-5ea1-45ba-8b70-0e5939eb1ecb)


<details>
<summary>Updates archive</summary>

nothing else yet

</details>

**Comment by heartacker** [comment]
Could you please provide a another text base panel， not only markdown  panel?

**Comment by zadjii-msft** [comment]
Sure, why not. Rendering plaintext is _easy_ comparatively to rendering markdown in WinUI 😜 

![fhl-markdown-open-text-file](https://github.com/microsoft/terminal/assets/18356694/b8ca820b-a768-438d-8174-c58a49be0af5)

**Comment by heartacker** [comment]
> Sure, why not. Rendering plaintext is _easy_ comparatively to rendering markdown in WinUI 😜 
> 
> ![fhl-markdown-open-text-file](https://github.com/microsoft/terminal/assets/18356694/b8ca820b-a768-438d-8174-c58a49be0af5)
> 

thank ，now i just need a short cut to send the  line command without text_selected ，or select_text
after ，we don't need copy->paste->commit line
